ray-curator/ray_curator/stages/audio/io/asr_inference.py

```diff
@@ -0,0 +1,143 @@
+
+    def setup(self) -> None:
+        """Initialise heavy object self.asr_model: nemo_asr.models.ASRModel"""
+        self.asr_model = nemo_asr.models.ASRModel.from_pretrained(model_name=self.model_name)
+
+    def inputs(self) -> tuple[list[str], list[str]]:
+        """Define the input attributes required by this stage.
+
+        Returns:
+            Tuple of (top_level_attrs, data_attrs) where:
+            - top_level_attrs: ["data"] - requires FileGroupTask.data to be populated
+        """
+        return ["data"], []
+
+    def outputs(self) -> tuple[list[str], list[str]]:
+        """Define the output attributes produced by this stage.
+
+        Returns:
+            Tuple of (top_level_attrs, data_attrs) where:
+            - top_level_attrs: ["data"] - populates FileGroupTask.data
+            - data_attrs: [self.filepath_key, self.text_key] - audiofile path and predicted text.
+        """
+        return ["data"], [self.filepath_key, self.text_key]
+
+    def transcribe(self, files: list[str]) -> list[str]:
+        """Run inference for speech recognition model
+         Args:
+            files: list of audio file paths.
+
+        Returns:
+            list of predicted texts.
+        """
+        outputs = self.asr_model.transcribe(files)
+        return [output.text for output in outputs]
+
+    def process_batch(self, tasks: list[FileGroupTask]) -> list[SpeechObject]:
+        """Process a audio task by reading audio file and do ASR inference.
+
+
+        Args:
+            tasks: List of FileGroupTask containing a path to audop file for inference.
+
+        Returns:
+            List of SpeechObject with self.filepath_key .
+            If errors occur, the task is returned with error information stored.
+        """
+        files = [task.data[0] for task in tasks]
+        outputs = self.transcribe(files)
+
+        audio_tasks = []
+        for i in range(len(outputs)):
+            text = outputs[i]
+            file_path = files[i]
+
+            entry = {self.filepath_key: file_path, self.text_key: text}
+
+            audio_task = SpeechObject(
+                task_id=f"{file_path}_task_id",
+                dataset_name=f"{self.model_name}_inference",
+                filepath_key=self.filepath_key,
+                data=entry,
+            )
+            audio_tasks.append(audio_task)
+        return audio_tasks
+
+    def process(self, task: FileGroupTask) -> list[SpeechObject]:
+        pass
+
+
+@dataclass
+class AsrNemoInference(CompositeStage[_EmptyTask, SpeechObject]):
+    """Composite stage that read audio files and do speech recognition inference using NeMo model.
+
+    This stage combines FilePartitioningStage and AsrNemoInferenceStage into a single
+    high-level operation for reading audiop files from a directory and processing
+    them.
+
+    Args:
+        input_audio_path: Path to the directory containing audio files
+        model_name: name of the speech recognition NeMo model
```

Comment:
I was wondering if there is a list of models supported here? Could you add a link?

Reply:
added to AsrNemoInferenceStage()